src/spares/oc_addon_template.lsl

```diff
@@ -75,62 +80,83 @@ UserCommand(integer iNum, string sStr, key kID) {
         
     }
 }
-Link(string packet, integer iNum, string sStr, key kID){
-    string pkt = llList2Json(JSON_OBJECT, ["pkt_type", packet, "iNum", iNum, "addon_name", g_sAddon, "bridge", FALSE, "sMsg", sStr, "kID", kID]);
-    if(g_kCollar!= "" && g_kCollar!= NULL_KEY) 
+
+Link(string packet, integer iNum, string sStr, key kID, list lLinkParams){
```

Comment:
Optin is a field unique to update and online packets only. It should just be a global list containing what the addon needs, then a fast check against packet type.

Reply:
Plan was to make the Link()-function future proof, as other calls *could* potentially get additional kv-pairs. If it's certain that those will only ever be needed by `update` and `online`, I'll change the code as requested.

Reply:
Yes- the only ones needing optin is update and online. The addon API as you see it in RC1 is final, it may change slightly in the future, but not in any breaking ways..